---
title: "gene list - step 03"
author: "Sharujan Suthakaran"
date: "01.Juni 2023"
---

```{r}
##junk
##trying to integrate achilles data info with .exp (until 4, but from where?)

View(prism.achilles.rm)

min(prism.achilles.rm) #-4,495

max(prism.achilles.rm) #4,874

View(prism.achilles.rm)

important_genes <- rownames(prism.achilles.breast[,which(prism.achilles. <= 0)]) # does not work

prism.achilles_mean <- apply(prism.achilles.breast, 2, mean)

important_genes <- colnames(prism.achilles.breast[,which(prism.achilles_mean >= 2)])

View(prism.achilles_mean)

View(important_genes)

important_gene_exp <- prism.exp[, colnames(prism.exp) %in% important_genes]

dim(important_gene_exp)

View(important_gene_exp)

important_gene_list <- colnames(important_gene_exp)

View(important_gene_list)

print(important_gene_list)

sum(which(prism.achilles_mean < -1))

sum(important_genes %in% colnames(prism.exp))


```



```{r}
min(prism.achilles.breast) #-2.749

max(prism.achilles.breast) #2.649

min(prism.achilles_mean) #-2.37806

max(prism.achilles_mean) #0.751765

hist(prism.achilles_mean)

important_genes <- colnames(prism.achilles.breast[,which(prism.achilles_mean >= 0)])

important_gene_exp <- prism.exp[, colnames(prism.exp) %in% important_genes]

dim(important_gene_exp)

important_gene_list <- colnames(important_gene_exp)

print(important_gene_list)

sum(important_genes %in% colnames(prism.exp))
```

https://www.gsea-msigdb.org/gsea/msigdb

**Hallmark gene sets (H): These gene sets represent well-defined biological states or processes that are coherently expressed. They can provide insights into common pathways and functions associated with breast cancer.**

Downloaded the gene sets as a JSON Bundle

```{r}

library(jsonlite)

hallmarks <- fromJSON("~/Downloads/h.all.v2023.1.Hs.json.txt")

View(hallmarks)

class(hallmarks)

```

*The hallmark gene set is a list divided into 50 different tumor influenced biological states*

**C6: oncogenic signature gene sets defined directly from microarray gene expression data from cancer gene perturbations.**

```{r}

C6.gene <- fromJSON("~/Downloads/c6.all.v2023.1.Hs.json.txt")

View(C6.gene)

```

**C8: cell type signature gene sets curated from cluster markers identified in single-cell sequencing studies of human tissue.**

```{r}

C8.cluster <- fromJSON("~/Downloads/c8.all.v2023.1.Hs.json.txt")

View(C8.cluster)

```

***C6 fits very good for our purpose, because it contains the genes for our filtration process and additionally has already clustered the genes by common function.***

C6 is a list containing 189 smaller lists named as abbreviations for different biological processes. These lists have different subrows with values, the important row: 'geneSymbols'

```{r}
lapply(C6.gene, function(x) x$geneSymbols)


C6.geneSym <- unlist(lapply(C6.gene, function(x) x$geneSymbols)) #single vector


View(C6.geneSym)

View(important_gene_list)

```

Looking for overlaps to proceed with filter.
```{r}

sum(important_gene_list %in% C6.geneSym) #3843


breast_gene <- intersect(C6.geneSym, important_gene_list)

print(breast_gene)


```

```{r}

View(prism.treat)



geneSearchEngine <- function(query) {
  matchingRows <- which(prism.treat$target == query)
  if (length(matchingRows) == 0) {
    cat("No matching treatment found.\n")
  } else {
    cat("Matching treatment(s):\n")
    for (row in matchingRows) {
      rowName <- row.names(prism.treat)[row]
      rowValue <- as.character(prism.treat$name[row])
      cat("Row Name:", rowName, "\tTreatment:", rowValue, "\n")
    }
  }
}


geneSearchEngine("PDF")



```


```{r}

treatmentSearchEngine <- function(query) {
  matchingRows <- which(prism.treat$name == query)
  if (length(matchingRows) == 0) {
    cat("No matching target found.\n")
  } else {
    cat("Matching target(s):\n")
    for (row in matchingRows) {
      rowName <- row.names(prism.treat)[row]
      rowValue <- as.character(prism.treat$target[row])
      cat("Row Name:", rowName, "\tTarget:", rowValue, "\n")
    }
  }
}


treatmentSearchEngine("noretynodrel")

sum(is.na(prism.treat$target))

```

***I will try to integrate cell line specific targets in order to get possible treatments for best response.***

```{r}
###manuell clustering

breast_carcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_carcinoma"),])$DepMap_ID

print(breast_carcinoma)

breast_ductal_carcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_ductal_carcinoma"),])$DepMap_ID

#breast_adenocarcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_adenocarcinoma"),])$DepMap_ID
```

```{r}
#prism.achilles.breast_adenocarcinoma <- prism.achilles.rm[which(rownames(prism.achilles.breast)%in%breast_adenocarcinoma),]

prism.achilles.breast_carcinoma <- prism.achilles.breast[which(rownames(prism.achilles.breast)%in%breast_carcinoma),]

prism.achilles.breast_ductal_carcinoma <- prism.achilles.breast[which(rownames(prism.achilles.breast)%in%breast_ductal_carcinoma),]

View(prism.achilles.breast_carcinoma)
```

**threshold for breast carcinoma subtype**
```{r}

imp.carcinoma <- apply(prism.achilles.breast_carcinoma, 2, mean)

#View(imp.carcinoma)

max(imp.carcinoma)

min(imp.carcinoma)

```

**threshold for breast ductal carcinoma subtype**
```{r}

imp.ductal_carcinoma <- apply(prism.achilles.breast_ductal_carcinoma, 2, mean)

#View(imp.ductal_carcinoma)

max(imp.ductal_carcinoma)

min(imp.ductal_carcinoma)

```

***integrate prism.snv for easier gene/cell line filter***
```{r}

PRISM.snv.breast <- prism.snv[prism.snv$DepMap_ID %in% breast_prism_cl$DepMap_ID,]

#View(PRISM.snv.breast)

prism.snv.breast <- PRISM.snv.breast[PRISM.snv.breast$Variant_Classification != "Silent",]

#View(prism.snv.breast)
```

*focus on damaging variants*
```{r}

prism.snv.breast_damage <- prism.snv.breast[prism.snv.breast$Variant_annotation == "damaging",]

View(prism.snv.breast_damage)

##for breast carcinoma

prism.snv.breast_carcinoma <- prism.snv.breast_damage[prism.snv.breast_damage$DepMap_ID %in% breast_carcinoma,]

##for breast ductal carcinoma

prism.snv.breast_ductal_carcinoma <- prism.snv.breast_damage[prism.snv.breast_damage$DepMap_ID %in% breast_ductal_carcinoma,]

View(prism.snv.breast_ductal_carcinoma)


###expression level, copy number and importance of these genes

View(breast_prism.exp) #cell line abbrev. are different

View(breast_prism.cnv)

View(prism.achilles.breast)
```