---
title: "gene list - step 03"
author: "Sharujan Suthakaran"
date: "01.Juni 2023"
---

# 1. threshold for gene importance

```{r}
##junk
##trying to integrate achilles data info with .exp (until 4, but from where?)

View(prism.achilles.rm)

min(prism.achilles.rm) #-4,495

max(prism.achilles.rm) #4,874

View(prism.achilles.rm)

important_genes <- rownames(prism.achilles.breast[,which(prism.achilles. <= 0)]) # does not work

prism.achilles_mean <- apply(prism.achilles.breast, 2, mean)

important_genes <- colnames(prism.achilles.breast[,which(prism.achilles_mean >= 2)])

View(prism.achilles_mean)

View(important_genes)

important_gene_exp <- prism.exp[, colnames(prism.exp) %in% important_genes]

dim(important_gene_exp)

View(important_gene_exp)

important_gene_list <- colnames(important_gene_exp)

View(important_gene_list)

print(important_gene_list)

sum(which(prism.achilles_mean < -1))

sum(important_genes %in% colnames(prism.exp))


```

```{r}
min(prism.achilles.breast) #-2.749

max(prism.achilles.breast) #2.649

min(prism.achilles_mean) #-2.37806

max(prism.achilles_mean) #0.751765

hist(prism.achilles_mean)

important_genes <- colnames(prism.achilles.breast[,which(prism.achilles_mean >= 0)])

important_gene_exp <- prism.exp[, colnames(prism.exp) %in% important_genes]

dim(important_gene_exp)

important_gene_list <- colnames(important_gene_exp)

print(important_gene_list)

sum(important_genes %in% colnames(prism.exp))
```

# 2. Biomarker gene sets

<https://www.gsea-msigdb.org/gsea/msigdb>

**Hallmark gene sets (H): These gene sets represent well-defined biological states or processes that are coherently expressed. They can provide insights into common pathways and functions associated with breast cancer.**

Downloaded the gene sets as a JSON Bundle

```{r}

library(jsonlite)

hallmarks <- fromJSON("~/Downloads/h.all.v2023.1.Hs.json.txt")

View(hallmarks)

class(hallmarks)

```

*The hallmark gene set is a list divided into 50 different tumor influenced biological states*

**C6: oncogenic signature gene sets defined directly from microarray gene expression data from cancer gene perturbations.**

```{r}

C6.gene <- fromJSON("~/Downloads/c6.all.v2023.1.Hs.json.txt")

View(C6.gene)

```

**C8: cell type signature gene sets curated from cluster markers identified in single-cell sequencing studies of human tissue.**

```{r}

C8.cluster <- fromJSON("~/Downloads/c8.all.v2023.1.Hs.json.txt")

View(C8.cluster)

```

***C6 fits very good for our purpose, because it contains the genes for our filtration process and additionally has already clustered the genes by common function.***

C6 is a list containing 189 smaller lists named as abbreviations for different biological processes. These lists have different subrows with values, the important row: 'geneSymbols'

```{r}
lapply(C6.gene, function(x) x$geneSymbols)


C6.geneSym <- unlist(lapply(C6.gene, function(x) x$geneSymbols)) #single vector


View(C6.geneSym)

View(important_gene_list)

```

Looking for overlaps to proceed with filter.

```{r}

sum(important_gene_list %in% C6.geneSym) #3843


breast_gene <- intersect(C6.geneSym, important_gene_list)

print(breast_gene)


```

# 3. Search Engine

***This is a gene Search Engine, which prints out treatments for the selected gene.***

```{r}
#FOR AARON


#install.packages("tidyverse")

library(tidyverse)

prism.treat_subset <- prism.treat %>%
  separate_rows(target, sep = ",") %>%
  
#print(prism.treat_subset)

#View(prism.treat_subset)

treat.gene_list <- prism.treat_subset[, c(1, 7)]

#View(treat.gene_list)

dim(treat.gene_list)
```

```{r}

#View(prism.treat)



geneSearchEngine <- function(query) {
  matchingRows <- which(prism.treat_subset$target == query)
  if (length(matchingRows) == 0) {
    cat("No matching treatment found.\n")
  } else {
    cat("Matching treatment(s):\n")
    for (row in matchingRows) {
      rowName <- row.names(prism.treat)[row]
      rowValue <- as.character(prism.treat$name[row])
      cat("Row Name:", rowName, "\tTreatment:", rowValue, "\n")
      if (rowName %in% rownames(breast_damage.achilles)) {
        achilles_value <- breast_damage.achilles[rowName, ]
        cat("Achilles Value:", achilles_value, "\n")
      } else {
        cat("Achilles Value: Not available\n")
      }
      
      if (rowName %in% rownames(breast_damage.exp)) {
        exp_value <- breast_damage.exp[rowName, ]
        cat("Exp Value:", exp_value, "\n")
      } else {
        cat("Exp Value: Not available\n")
      }
      
      if (rowName %in% rownames(breast_damage.cnv)) {
        cnv_value <- breast_damage.cnv[rowName, ]
        cat("Cnv Value:", cnv_value, "\n")
      } else {
        cat("Cnv Value: Not available\n")
      }
    }
  }
}

geneSearchEngine("GOLGA4")


#intersect(colnames(breast_damage.achilles), colnames(breast_damage.cnv))
```

***This function undergoes printing out values of .exp/.achilles and.cnv of damaging genes. Additionally, it prints possible treatments, if the gene is a target in prism.treat.***

```{r}

for (gene_name in prism.snv.breast_damage$Hugo_Symbol) {
   
    matchingRows <- which(prism.treat_subset$target == gene_name)
    if (length(matchingRows) == 0) {
      cat("No matching treatment found for Gene:", gene_name, "\n")
    } else {
      cat("Matching treatment(s) for Gene:", gene_name, "\n")
      for (row in matchingRows) {
        rowName <- row.names(prism.treat)[row]
        rowValue <- as.character(prism.treat$name[row])
        cat("Row Name:", rowName, "\tTreatment:", rowValue, "\n")
      }
    }
    
    
    if (gene_name %in% colnames(breast_damage.exp) &&
        gene_name %in% colnames(breast_damage.achilles) &&
        gene_name %in% colnames(breast_damage.cnv)) {
      
      
      exp_value <- breast_damage.exp[rownames(breast_damage.exp), gene_name]
      cat("Expression Level:", exp_value, "\n")
      
      
      achilles_value <- breast_damage.achilles[rownames(breast_damage.achilles), gene_name]
      cat("Achilles Value:", achilles_value, "\n")
      
    
      cnv_value <- breast_damage.cnv[rownames(breast_damage.cnv), gene_name]
      cat("Copy Number:", cnv_value, "\n")
      
      cat("\n")
    } else {
      cat("Gene:", gene_name, "is not present in all datasets.\n\n")
      
    if (gene_name %in% colnames(breast_damage.exp)) {
      exp_value <- breast_damage.exp[rownames(breast_damage.exp), gene_name]
      cat("Expression Level (breast_damage.exp):\n")
      for (i in 1:length(rownames(breast_damage.exp))) {
      cat(rownames(breast_damage.exp)[i], ":", exp_value[i], "\n")
}
    }
    
    if (gene_name %in% colnames(breast_damage.achilles)) {
      achilles_value <- breast_damage.achilles[rownames(breast_damage.achilles), gene_name]
      cat("Achilles Value (breast_damage.achilles):", achilles_value, "\n")
    }
    
    if (gene_name %in% colnames(breast_damage.cnv)) {
      cnv_value <- breast_damage.cnv[rownames(breast_damage.cnv), gene_name]
      cat("Copy Number (breast_damage.cnv):", rownames(breast_damage.cnv),":",cnv_value, "\n")
    }
    
    cat("\n")
  }
}

```

***Search engine for treatments, needs further improvement***

```{r}

treatmentSearchEngine <- function(query) {
  matchingRows <- which(prism.treat$name == query)
  if (length(matchingRows) == 0) {
    cat("No matching target found.\n")
  } else {
    cat("Matching target(s):\n")
    for (row in matchingRows) {
      rowName <- row.names(prism.treat)[row]
      rowValue <- as.character(prism.treat$target[row])
      cat("Row Name:", rowName, "\tTarget:", rowValue, "\n")
    }
  }
}


treatmentSearchEngine("noretynodrel")

sum(is.na(prism.treat$target))

```

# 4. Manuell clustering

***I will try to integrate cell line specific targets in order to get possible treatments for best response.***

```{r}
###manuell clustering

breast_carcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_carcinoma"),])$DepMap_ID

print(breast_carcinoma)

breast_ductal_carcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_ductal_carcinoma"),])$DepMap_ID

#breast_adenocarcinoma <- c(breast_prism_cl[which(breast_prism_cl$lineage_subtype == "breast_adenocarcinoma"),])$DepMap_ID
```

```{r}
#prism.achilles.breast_adenocarcinoma <- prism.achilles.rm[which(rownames(prism.achilles.breast)%in%breast_adenocarcinoma),]

prism.achilles.breast_carcinoma <- prism.achilles.breast[which(rownames(prism.achilles.breast)%in%breast_carcinoma),]

prism.achilles.breast_ductal_carcinoma <- prism.achilles.breast[which(rownames(prism.achilles.breast)%in%breast_ductal_carcinoma),]

View(prism.achilles.breast_carcinoma)
```

**threshold for breast carcinoma subtype**

```{r}

imp.carcinoma <- apply(prism.achilles.breast_carcinoma, 2, mean)

#View(imp.carcinoma)

max(imp.carcinoma)

min(imp.carcinoma)

```

**threshold for breast ductal carcinoma subtype**

```{r}

imp.ductal_carcinoma <- apply(prism.achilles.breast_ductal_carcinoma, 2, mean)

#View(imp.ductal_carcinoma)

max(imp.ductal_carcinoma)

min(imp.ductal_carcinoma)

```

# 5. damaging Genes

***integrate prism.snv for easier gene/cell line filter***

```{r}

PRISM.snv.breast <- prism.snv[which(as.character(prism.snv$DepMap_ID) %in% breast_prism_cl$DepMap_ID),]

#View(PRISM.snv.breast)

prism.snv.breast <- PRISM.snv.breast[PRISM.snv.breast$Variant_Classification != "Silent",]

#View(prism.snv.breast)
```

*focus on damaging variants*

```{r}

prism.snv.breast_damage <- prism.snv.breast[prism.snv.breast$Variant_annotation == "damaging",]

#View(prism.snv.breast_damage)

##for breast carcinoma

prism.snv.breast_carcinoma <- prism.snv.breast_damage[prism.snv.breast_damage$DepMap_ID %in% breast_carcinoma,]

##for breast ductal carcinoma

prism.snv.breast_ductal_carcinoma <- prism.snv.breast_damage[prism.snv.breast_damage$DepMap_ID %in% breast_ductal_carcinoma,]

#View(prism.snv.breast_ductal_carcinoma)


###expression level, copy number and importance of these genes

#View(breast_prism.exp) #cell line abbrev. are different

#View(breast_prism.cnv)

#View(prism.achilles.breast)
```

```{r}


missingrows.achilles <- matrix(0, nrow = 5, ncol = ncol(prism.achilles.breast))

colnames(missingrows.achilles) <- colnames(prism.achilles.breast)

rownames(missingrows.achilles) <- setdiff(rownames(breast_prism.exp), rownames(prism.achilles.breast))[1:5]

#View(missingrows.achilles)

breast_prism.achilles <- rbind(prism.achilles.breast, missingrows.achilles)

#View(breast_prism.achilles)

dim(breast_prism.achilles)


##attach damage genes as columns

colnames(breast_prism.exp)%in%colnames(breast_prism.achilles) #different column order

###damage gene data frame for .exp

damage.exp <- breast_prism.exp[,colnames(breast_prism.exp) %in% prism.snv.breast_damage$Hugo_Symbol]

exp.sorted_col <- colnames(damage.exp)[order(colnames(damage.exp))]

breast_damage.exp <- damage.exp[, exp.sorted_col]

#View(breast_damage.exp)


###damage gene data frame for .cnv

damage.cnv <- breast_prism.cnv[,colnames(breast_prism.achilles) %in% prism.snv.breast_damage$Hugo_Symbol]

cnv.sorted_col <- colnames(damage.cnv)[order(colnames(damage.cnv))]

breast_damage.cnv <- damage.cnv[, cnv.sorted_col]

###damage gene data frame for .achilles

breast_damage.achilles <- as.data.frame(breast_prism.achilles[,colnames(breast_prism.achilles) %in% prism.snv.breast_damage$Hugo_Symbol])

damage.achilles_mean <- apply(breast_damage.achilles, 2, mean)

min(breast_damage.achilles)

max(damage.achilles_mean)

order(colnames(breast_damage.achilles)) #in perfect order

```

```{r}

############################################################################################
important_genes <- colnames(breast_damage.achilles[,which(damage.achilles_mean <= -0.5)])

hist(damage.achilles_mean)

important_gene_exp <- prism.exp[, colnames(prism.exp) %in% important_genes]

dim(important_gene_exp)

important_gene_list <- colnames(important_gene_exp)

#print(important_gene_list)


```
```{r}
carcinoma_damage.achilles <- as.data.frame(breast_prism.achilles[rownames(prism.achilles.breast_carcinoma),colnames(breast_prism.achilles) %in% prism.snv.breast_damage$Hugo_Symbol])

#View(carcinoma_damage.achilles)

carcinoma.damage.achilles_mean <- apply(carcinoma_damage.achilles, 2, mean)

min(carcinoma.damage.achilles_mean)

max(carcinoma.damage.achilles_mean)

#################

important_carcinoma.genes <- colnames(carcinoma_damage.achilles[,which(carcinoma.damage.achilles_mean <= -0.5)])

important_carcinoma.gene_exp <- prism.exp[, colnames(prism.exp) %in% important_carcinoma.genes]

dim(important_carcinoma.gene_exp)

important_carcinoma.gene_list <- colnames(important_carcinoma.gene_exp)

#print(important_carcinoma.gene_list)

```

```{r}
ductal_carcinoma_damage.achilles <- as.data.frame(breast_prism.achilles[rownames(prism.achilles.breast_ductal_carcinoma),colnames(breast_prism.achilles) %in% prism.snv.breast_damage$Hugo_Symbol])

#View(carcinoma_damage.achilles)

ductal_carcinoma.damage.achilles_mean <- apply(ductal_carcinoma_damage.achilles, 2, mean)

min(ductal_carcinoma.damage.achilles_mean)

max(ductal_carcinoma.damage.achilles_mean)

########################
important_ductal.genes <- colnames(ductal_carcinoma_damage.achilles[,which(ductal_carcinoma.damage.achilles_mean <= -0.5)])

important_ductal.gene_exp <- prism.exp[, colnames(prism.exp) %in% important_ductal.genes]

dim(important_ductal.gene_exp)

important_ductal.gene_list <- colnames(important_ductal.gene_exp)

#print(important_carcinoma.gene_list)
```


```{r}
achilles.cnv.exp_breast <- data.frame(row.names = rownames(breast_prism.exp), stringsAsFactors = FALSE)

for (gene_name in colnames(breast_damage.achilles)) {
  if (gene_name %in% colnames(breast_damage.exp) && gene_name %in% colnames(breast_damage.cnv)) {
  achilles_value <- breast_damage.achilles[, gene_name]
  exp_value <- breast_damage.exp[, gene_name]
  cnv_value <- breast_damage.cnv[, gene_name]
  achilles.cnv.exp_breast <- cbind(achilles.cnv.exp_breast, achilles_value, exp_value, cnv_value)
}
else {
    warning(paste("Column", gene_name, "does not exist in breast_damage.exp. Skipping."))
  }
}

View(achilles.cnv.exp_breast)

# Rename the columns
colnames(achilles.cnv.exp_breast) <- c("Gene_Name", paste0("Achilles_", colnames(breast_damage.achilles)), paste0("Exp_", colnames(breast_damage.exp)))


```

```{r}
#achilles.cnv.exp_breast <- data.frame(row.names = rownames(breast_prism.exp), stringsAsFactors = FALSE)

#for (gene_name in colnames(breast_damage.achilles)) {
  #if (gene_name %in% colnames(breast_damage.exp) && gene_name %in% colnames(breast_damage.cnv)) {
  #achilles_value <- breast_damage.achilles[, gene_name]
  #exp_value <- breast_damage.exp[, gene_name]
  #cnv_value <- breast_damage.cnv[, gene_name]

  #gene_values <- data.frame(breast_damage.achilles = achilles_value, breast_damage.exp = exp_value, breast_damage.cnv = cnv_value)
  #achilles.cnv.exp_breast <- cbind(achilles.cnv.exp_breast, gene_values)
  }
}


#View(achilles.cnv.exp_breast)


View(prism.treat)

intersect(colnames(prism.exp), prism.treat$target)


```
